src/crystal/system/unix/process.cr

```diff
@@ -0,0 +1,252 @@
+require "c/signal"
+require "c/stdlib"
+require "c/sys/resource"
+require "c/unistd"
+
+class Process
+  protected def open_system
+    exists_system?
+  end
+
+  protected def wait_system
+    Crystal::SignalChildHandler.wait(@waitpid, @pid)
+  end
+
+  protected def self.exit_system(status = 0) : NoReturn
+    LibC.exit(status)
+  end
+
+  protected def self.pid_system : Int64
+    LibC.getpid.to_i64
+  end
+
+  protected def self.ppid_system : Int64
+    LibC.getppid.to_i64
+  end
+
+  protected def exists_system?
```

Comment:
AFAIK we don't need the `#exists_system?` and `#pgid_system` instance methods, even on Windows. The class methods are sufficient and `Process.exists?(pid)` and `Process.pgid(pid)` can directly call both system class methods. It will avoid a few indirections.

Reply:
I prefer to keep them to be able to reuse process handle in Windows implementation when working with process instance (on Win, you need open handle to the process for most of the methods).